### DISCUSSION TITLE
How to show output in integrated terminal instead of debug console?

### DISCUSSION THREAD

**Discussion by ssinzdaryl** [microsoft/vscode-cpptools#10116] [Q&A] [answered]
## Environment

OS and version: M1 macOS 13.0
VS Code: 1.73.0
C/C++ extension: v1.12.4

Can't show output in integrated terminal instead of debug console, even if I set "avoidWindowsConsoleRedirection": to false and add "externalConsole": false, but the output still show in debug console.

```
// my launch.json

{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "C/C++: clang++ build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/output/${fileBasenameNoExtension}",
            "args": [],
            "cwd": "${workspaceFolder}",
            "environment": [],
            "MIMode": "lldb",
            "preLaunchTask": "C/C++: clang++ build active file",

            "avoidWindowsConsoleRedirection": false,
            "externalConsole": false,
        }
    ]
}
```

<img width="702" alt="image" src="https://user-images.githubusercontent.com/102724735/200099499-1f4df4b2-ee8a-47d4-a72d-c70eed7c6209.png">

**Comment by macTracyHuang** [comment]
> ## Environment
> OS and version: M1 macOS 13.0 VS Code: 1.73.0 C/C++ extension: v1.12.4
> 
> Can't show output in integrated terminal instead of debug console, even if I set "avoidWindowsConsoleRedirection": to false and add "externalConsole": false, but the output still show in debug console.
> 
> ```
> // my launch.json
> 
> {
>     "version": "0.2.0",
>     "configurations": [
>         {
>             "name": "C/C++: clang++ build and debug active file",
>             "type": "cppdbg",
>             "request": "launch",
>             "program": "${fileDirname}/output/${fileBasenameNoExtension}",
>             "args": [],
>             "cwd": "${workspaceFolder}",
>             "environment": [],
>             "MIMode": "lldb",
>             "preLaunchTask": "C/C++: clang++ build active file",
> 
>             "avoidWindowsConsoleRedirection": false,
>             "externalConsole": false,
>         }
>     ]
> }
> ```
> 
> <img alt="image" width="702" src="https://user-images.githubusercontent.com/102724735/200099499-1f4df4b2-ee8a-47d4-a72d-c70eed7c6209.png">

Hey, I found a workaround:
Add "internalConsoleOptions": "openOnSessionStart" in the launch.json can automatically switch to DEBUG CONSOLE after running the file.

**Comment by ssinzdaryl** [comment, marked as answer]
@macTracyHuang

Thanks bro, it works, but if you want to show the output in integrated terminal, i found a better way.

1. Install a extension named CodeLLDB, a native debugger powered by LLDB.
2. Modify the "type" property in the "launch.json", change "cppdbg" to "lldb".
3. You will find some property in launch.json not allowed, remove them.

Then the output can show in integrated terminal.

```
// launch.json

{
    "configurations": [
        {
            "name": "C/C++: clang++ build and debug active file",
            "type": "lldb",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}",
            "args": [],
            "cwd": "${fileDirname}",
            "preLaunchTask": "C/C++: clang++ build active file",
        }
    ],
    "version": "2.0.0"
}
```

**Reply by dikshant79** [reply]
thanks a lot bro
